linera-service/src/util.rs

```diff
@@ -2,52 +2,94 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use anyhow::{bail, Context as _, Result};
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 use tokio::process::Command;
 use tracing::{debug, error};
 
 /// Attempts to resolve the path and test the version of the given binary against our
-/// package version. This is meant for binaries of the Linera repository.
+/// package version.
+///
+/// This is meant for binaries of the Linera repository. We use the current running binary
+/// to locate the parent directory where to look for the given name.
 pub async fn resolve_binary(name: &'static str, package: &'static str) -> Result<PathBuf> {
-    debug!("Resolving binary {name} based on the current binary path.");
-    let mut current_binary_path = PathBuf::from(
+    let current_binary = PathBuf::from(
         std::env::args()
             .next()
             .expect("args should start with the current binary path"),
-    )
-    .canonicalize()?;
-    current_binary_path.pop();
+    );
```

Comment:
Should we use [`env::current_exe`](https://doc.rust-lang.org/std/env/fn.current_exe.html) instead? If I understand correctly, that should return the full path.

Reply:
Nice! Let me have a look.